### PR TITLE
Built out the Edit Gas Price UI's

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Built out the Edit Gas Price UI's for both `eip1559` and non `eip1559`

Note: Will be wired up to the `ethtxcontroller` API in a future PR here https://github.com/brave/brave-browser/issues/18370

Resolves <https://github.com/brave/brave-browser/issues/18069>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

Non `eip1559`:

https://user-images.githubusercontent.com/40611140/134835501-3e3ff029-b75c-4228-a414-885c83813b0a.mov

`eip1559`:

https://user-images.githubusercontent.com/40611140/134945225-48111301-1d91-4eec-8599-02e82bd2056f.mov

### CODE REVIEW

**Review by onyb — APPROVED** [review]
Looks good! Approved with minor comments.

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx [hunk 1]
```diff
@@ -173,112 +174,135 @@ function ConfirmTransactionPanel (props: Props) {
...
+              onSubmit={onSelectTab('transaction')}
+              text='Transaction'
+            />
+            <PanelTab
+              isSelected={selectedTab === 'details'}
+              onSubmit={onSelectTab('details')}
+              text='Details'
+            />
+          </TabRow>
+          <MessageBox isDetails={selectedTab === 'details'} isApprove={transactionInfo.txType === 2}>
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx]: Changed!

File: components/brave_wallet_ui/components/extension/edit-gas/index.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+  const handleGasPriceInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGasPrice(event.target.value)
+  }
+
+  const handleGasAmmountLimitInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGasAmountLimit(event.target.value)
+  }
+
+  const handleGasTipAmmountLimitInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/edit-gas/index.tsx]: Nice find! Fixed!

File: components/brave_wallet_ui/components/extension/edit-gas/index.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+  // Will determine how best to differentiate between EIP1559 and Legacy
+  // when wiring up to API.
+  const isEIP1559 = false
+
+  const handleGasPriceInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGasPrice(event.target.value)
+  }
+
+  const handleGasAmmountLimitInputChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/edit-gas/index.tsx]: Fixed!

File: components/brave_wallet_ui/components/extension/edit-gas/index.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    selectedNetwork,
+    transactionInfo
+  } = props
+  const { txData } = transactionInfo
+  const { baseData } = txData
+  const [maxPriorityPanel, setMaxPriorityPanel] = React.useState<MaxPriorityPanels>(MaxPriorityPanels.setSuggested)
+  const [suggestedMaxPriorityFee, setSuggestedMaxPriorityFee] = React.useState<string>(suggestedMaxPriorityFees[1])
+  const [suggestedSliderStep, setSuggestedSliderStep] = React.useState<string>('1')
+  const [gasPrice, setGasPrice] = React.useState<string>(baseData.gasPrice)
+  const [gasAmountLimit, setGasAmountLimit] = React.useState<string>(baseData.gasLimit)
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/edit-gas/index.tsx]: Done!

**Pull request merged**